src/main/java/com/redhat/cloud/notifications/auth/RbacIdentityProvider.java

```diff
@@ -20,13 +21,23 @@
     @RestClient
     RbacServer rbacServer;
 
+    @ConfigProperty(name = "rbac.enabled", defaultValue = "true")
+    Boolean isRbacEnabled;
+
     @Override
     public Class<RhIdentityAuthenticationRequest> getRequestType() {
         return RhIdentityAuthenticationRequest.class;
     }
 
     @Override
     public Uni<SecurityIdentity> authenticate(RhIdentityAuthenticationRequest rhAuthReq, AuthenticationRequestContext authenticationRequestContext) {
+        if (!isRbacEnabled) {
+            return Uni.createFrom().item(() -> QuarkusSecurityIdentity.builder()
+                    .addRole("read")
```

Comment:
I think you need to add a dummy principal here, as otherwise `principal.getAccount()` within the routers may NPE.

Reply:
This method returns SecurityIdentity and not Principal?